tvtk/tools/visual.py

```diff
@@ -763,38 +760,36 @@ class Ring(HasTraits):
     def __init__(self, **traits):
         self.property = self.actor.property
 
-        HasTraits.__init__(self, **traits)
-
-        self._create_points()
-        self._pos_changed(numpy.array([0.0, 0.0, 0.0]), self.pos)
-        self._x_changed(self.x)
-        self._y_changed(self.y)
-        self._z_changed(self.z)
-        self._color_changed(self.color)
-        self._visibility_changed(self.visibility)
-        self._thickness_changed(self.thickness)
-        self._axis_changed(numpy.array((1.0, 0.0, 0.0)), self.axis)
-
         self.normals = tvtk.PolyDataNormals()
-        configure_input_data(self.normals, self.polydata)
-        configure_input_data(self.tube, self.normals.outputs)
+        configure_input(self.normals, self.polydata)
+        configure_input(self.tube, self.normals)
         self.normals.update()
         self.tube.update()
         self.tube.number_of_sides = 4
         self.tube.capping = 1
 
         m = tvtk.PolyDataMapper()
-        configure_input_data(m, self.tube.outputs)
+        configure_input(m, self.tube)
         self.actor.mapper = m
         self.property = self.actor.property
         self.property.representation = self.representation
-        show_actor(self.actor)
-        self.viewer = get_viewer()
 
+        self.viewer = get_viewer()
         self.property.on_trait_change(self.viewer.scene.render)
         self.actor.on_trait_change(self.viewer.scene.render)
         self.tube.on_trait_change(self.viewer.scene.render)
 
+        HasTraits.__init__(self, **traits)
```

Comment:
No particular reason.  I wrote the tests and got it all working and fixed the issues.  This can be done later if someone cares or has the time.